lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -233,9 +233,9 @@ var doPersist =  function() {
     that.emit('beforePlaylistChanged', that);
 
     that.isDirty = false;
-    if(that.playlistImp.inner.liveWindowDuration) {
-        that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
-    }
+    that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
+    that.playlistImp.inner.expirationTime = that.playlistImp.inner.presentationEndTime;
```

Comment:
yes. the business logic for the 2 values is implemented in crossDCsStreamPersistence.js, per your request the PlaylistGenrator is not involved.